### ISSUE TITLE
Request WithArguments - no support for repeated arguments in query string

### ISSUE THREAD

**Issue by ygorccardoso** [Pathoschild/FluentHttpClient#70]
When we have API's with query string parameters that are enumerables/lists, the standard way to specify them in the request is by repeating the parameter in the query string with different values; eg.:

`http://api.com/Movies?genre=action&genre=drama`

If we try to call WithArguments with a KeyValuePair<string, string> with the values of the example above, it throws an exception of duplicated key, since internally this is converted to a dictionary.

A workaround is doing a foreach of this and calling WithArgument for each one individually.

**Comment by Pathoschild** [comment]
Hi @ygorccardoso. Can you provide a code example of the failing WithArguments call?

**Comment by Jericho** [comment]
I can see @ygorccardoso's point. The `WithArguments(object arguments)` method in Request.cs invokes `GetArguments` to convert the object into a `IDictionary<string, object>`. This prevents situations where we need multiple querystring arguments with the same name since the name of the arguments is used as the key in the dictionary.

A suggestion would be for the `GetArguments` method to return `IEnumerable<KeyValuePair<string, object>>` which would allow multiple arguments with same name and would enable use cases such as:

```csharp
var apiClient = new FluentClient("http://api.com");
var response = apiClient
    .GetAsync("Movies")
    .WithArguments(new[]
    {
        new KeyValuePair<string, object>("genre", "action"),
        new KeyValuePair<string, object>("genre", "drama")
    })
    .AsMessage();
```

In the mean time, the workaround @ygorccardoso suggests is perfectly valid (but can be tedious if the list of parameters is quite long): call `WithParameter` for each parameter like so:

```csharp
var apiClient = new FluentClient("http://api.com");
var response = apiClient
    .GetAsync("Movies")
    .WithArgument("genre", "action")
    .WithArgument("genre", "drama")
    .AsMessage();
```

**Comment by Jericho** [comment]
One more observation: the method in `UriExtensions.cs` that ultimately converts the KeyValuePair items into query string already supports multiple values with the same key (notice it accepts an array instead of a dictionary):

```csharp
public static Uri WithArguments(this Uri uri, params KeyValuePair<string, object>[] arguments)
```

so it's weird that `WithArguments(object arguments)` would not permit this scenario.

**Comment by Pathoschild** [comment]
Yep, it's an easy fix. I just hadn't accounted for someone passing actual key/value pairs into `WithArguments`.

**Comment by Jericho** [comment]
I will have a PR momentarily.

**Comment by ygorccardoso** [comment]
@Jericho thanks, that's great, those were exactly my thoughts, opened the ticket to do more or less the same but you were too quick 👍

**Comment by Pathoschild** [comment]
@ygorccardoso Fixed in the upcoming release via #71. :)
